components/movementsensor/wheeledodometry/wheeled_odometry.go

```diff
@@ -0,0 +1,302 @@
+// Package wheeledodometry implements an odometery estimate from an encoder wheeled base
+package wheeledodometry
+
+import (
+	"context"
+	"errors"
+	"math"
+	"sync"
+	"time"
+
+	"github.com/edaniels/golog"
+	"github.com/golang/geo/r3"
+	geo "github.com/kellydunn/golang-geo"
+	"go.viam.com/utils"
+
+	"go.viam.com/rdk/components/base"
+	"go.viam.com/rdk/components/motor"
+	"go.viam.com/rdk/components/movementsensor"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/spatialmath"
+	rdkutils "go.viam.com/rdk/utils"
+)
+
+var (
+	model     = resource.DefaultModelFamily.WithModel("wheeled-odometry")
+	lastLeft  = 0.0
+	lastRight = 0.0
+)
+
+const defaultTimeInt = 500
+
+// Config is used for converting a wheeledodometry MovementSensor config attributes.
+type Config struct {
+	LeftMotors   []string `json:"left_motors"`
+	RightMotors  []string `json:"right_motors"`
+	Base         string   `json:"base"`
+	TimeInterval float64  `json:"time_interval,omitempty"`
+}
+
+type motorPairs struct {
+	left  motor.Motor
+	right motor.Motor
+}
+
+type odometry struct {
+	resource.Named
+	resource.AlwaysRebuild
```

Comment:
:exploding_head: I totally missed this, but during reconfiguration, we'll reset our position to (0,0), rather than wherever the robot had been before you changed the config file. Is that a problem? If so, it's not the end of the world to fix, but it does seem annoying: we'd need to take, like, half of `newWheeledOdometry` and put it into `Reconfigure`, and then have the former call the latter.

Reply:
Adding reconfigure is a part of the epic but not this ticket